### PR TITLE
Replacing DatabaseUtil for Jdbc. BankAccount Controller

### PR DESCRIPTION
**Pull request by thok54**

### CODE REVIEW

File: src/main/java/com/banking/bankservice/repository/MySqlBankRepository.java [hunk 1]
```diff
@@ -38,95 +32,58 @@ public Bank mapRow(ResultSet rs, int rowNum) throws SQLException {
...
-            if (rs.next()) {
-                String name = rs.getString("name");
-                String address = rs.getString("address");
-                return new Bank(id, name, address);
-            }
-            throw new EntityNotFoundException(String.format("Bank with ID = %d does not exist", id));
+            return jdbcTemplate.queryForObject(String.format(
+                    "select * from BANKS where id = %d", id), new RowMapper<Bank>() {
+                @Override
+                public Bank mapRow(ResultSet rs, int i) throws SQLException {
```
Comment (jantpedraza) [inline src/main/java/com/banking/bankservice/repository/MySqlBankRepository.java]: This method here (mapRow) is duplicated in this class. Please, create a class for it.

File: src/main/java/com/banking/bankservice/repository/MySqlBankRepository.java [hunk 2]
```diff
@@ -38,95 +32,58 @@ public Bank mapRow(ResultSet rs, int rowNum) throws SQLException {
...
-
-        } catch (SQLException e) {
-            e.printStackTrace();
-        } finally {
-            dataBaseUtil.closeConections(con);
-        }
+        String name = bank.getName();
+        String address = bank.getAddress();
+        jdbcTemplate.update(String.format(
+                "INSERT INTO BANKS (name, address) VALUES (\"%s\",\"%s\")", name, address));
```
Comment (jantpedraza) [inline src/main/java/com/banking/bankservice/repository/MySqlBankRepository.java]: `jdbcTemplate.update()` method accepts the query with parameters so you can avoid the `String.format()`.

For instance, that code could be replaced by:

```
jdbcTemplate.update("INSERT INTO BANKS (name, address) VALUES (?, ?)", name, address);
```

File: src/main/java/com/banking/bankservice/web/BankAccountController.java [hunk 3]
```diff
@@ -16,9 +19,44 @@
     private RestTemplate restTemplate;
 
     @GetMapping
-    public Account get() {
-        ResponseEntity<Account> response = restTemplate.getForEntity("http://localhost:8090/account", Account.class);
+    public Account getAll() {
+        ResponseEntity<Account> response = restTemplate.getForEntity("http://localhost:8085/account", Account.class);
```
Comment (jantpedraza) [inline src/main/java/com/banking/bankservice/web/BankAccountController.java]: The string `http://localhost:8085/account` is duplicated 7 times! What would clean code say? :P

**Review by jantpedraza — CHANGES_REQUESTED** [review]

File: src/main/java/com/banking/bankservice/web/BankAccountController.java (hunk 3 above)
Comment (thok54) [reply src/main/java/com/banking/bankservice/web/BankAccountController.java]: Done at last commit

File: src/main/java/com/banking/bankservice/repository/MySqlBankRepository.java (hunk 1 above)
Comment (thok54) [reply src/main/java/com/banking/bankservice/repository/MySqlBankRepository.java]: Done at last commit as private method within same class

File: src/main/java/com/banking/bankservice/repository/MySqlBankRepository.java (hunk 2 above)
Comment (thok54) [reply src/main/java/com/banking/bankservice/repository/MySqlBankRepository.java]: Done, but seems to work only for jdbcTemplate.update()

**Pull request merged**